### DISCUSSION TITLE
Some fragments can't convert SMARTS to SIMLES.

### DISCUSSION THREAD

**Discussion by yilin667612** [rdkit/rdkit#4697] [Q&A]
Hi everyone!
I found some fragments can't convert SMARTS to SIMLES in 'Data/FragmentDescriptors.csv'.

Configuration:
RDKit Version: 2021.09.2 
Operating system: Linux
Python version: 3.7

for example:
fr_oxime	"Number of oxime groups"	[CX3]=[NX2]-[OX2]	C=N-O
fr_azide	"Number of azide groups"	[$(*-[NX2-]-[NX2+]#[NX1]),$(*-[NX2]=[NX2+]=[NX1-])]	[#6]-N=[N+]=[N-]	

Input:
```
import rdkit 
from rdkit import Chem 
m1 = Chem.MolFromSmarts("[CX3]=[NX2]-[OX2]")  
m2 = Chem.MolFromSmarts("[$(*-[NX2-]-[NX2+]#[NX1]),$(*-[NX2]=[NX2+]=[NX1-])]")   
print(m1,m2)  
print("m1:",Chem.MolToSmiles(m1))  
print("m2:",Chem.MolToSmiles(m2)) 
```

Output:
```
<rdkit.Chem.rdchem.Mol object at 0x7f06974c4710> <rdkit.Chem.rdchem.Mol object at 0x7f06974c4440>  
m1: C=NO  
m2: *
```

My question is why SIMLES of m2 is '*' ?

**Comment by dehaenw** [comment]
Hello @yilin667612 ,
I think the reason why it fails is because the fragments is a recursive SMARTS, see p.21 of the SMARTS manual https://www.daylight.com/dayhtml/doc/theory/index.pdf this is done because there are several resonance forms of the azide fragment and the SMARTS query needs to match only one of the two. for many SMARTS patterns including non recursive ones there is generally not a single SMILES available as SMARTS can contain far more general pattern definitions.

**Reply by yilin667612** [reply]
I figured it out. Thank you for the quick reply!
